crates/symbolicator/src/services/download/gcs.rs

```diff
@@ -260,6 +263,18 @@ impl GcsDownloader {
         }
     }
 
+    pub async fn download_source(
+        &self,
+        file_source: GcsRemoteDif,
+        destination: PathBuf,
+    ) -> Result<Result<DownloadStatus, DownloadError>, Elapsed> {
```

Comment:
I wonder if we can keep this simpler by returning `DownloadError::Canceled` here instead of adding another layer or Result with an Elapsed Err variant.  That's what it turns into anyway, and I don't think it loses any information?